how-it-works/data-and-provider-model/data-flow-and-lifecycle.md

```diff
@@ -1,37 +1,163 @@
+The first steps of the dataflow are as follows:
 
-5. **Retrieve the file**: When you request the file, the MSP returns the bytes plus a small cryptographic proof (Merkle proof). Your app automatically checks it against the file's on-chain commitment (anchored in the bucket), so you know the content matches the committed content. Files can also be shared via generated links with optional expiration and password protection, as described above.
+```mermaid
+graph LR
+    A[Connect wallet<br>or app identity]
+    B[Select<br>an MSP]
+    C[Create or reuse bucket<br>to hold files]
+    D[Set policy for desired<br>BSP replicas]
+    A-->B
+    B-->C
+    C-->D
+```
 
-6. **Ongoing health checks**: On a schedule, the chain challenges BSPs (not MSPs) to prove they still hold your data. BSPs provide small cryptographic proofs; slashing is possible in the event of data loss by BSPs per the protocol rules.
+## Upload the File
 
-7. **Update or replace a file**: Uploading changed content produces a new fingerprint (a new version). The bucket updates its on-chain summary to reference the new version. Whether older versions are kept is up to your app or storage provider policy.
+Once you connect to the network, select your MSP, and have a bucket, the upload steps are as follows:
+
+```mermaid
+sequenceDiagram
+    participant User/App
+    participant MSP
+    participant Offchain Storage
+    participant BSP
+    participant Blockchain
+
+    User/App->>MSP: Open a storage request<br>for your bucket
+    MSP->>Offchain Storage: Accept storage request,<br>Store the bytes off-chain
+    MSP->>BSP: Coordinate replications<br>for backup copies
+    BSP->>MSP: Confirm creation<br>of replica copies
+    MSP->>Blockchain: Anchor Merkle-root<br>of bucket update
+```
+
+If replication or anchoring fails, the upload aborts.
+
+## Share Your Bucket
+
+You can share access to your bucket by generating a link that provides view-only or view-and-upload access to its contents, including all files and folders.
+
+## Retrieve the File
+
+The process to retrieve a file you successfully uploaded to DataHaven is as follows:
+
+```mermaid
+sequenceDiagram
+    participant User/App
+    participant MSP
+    participant Blockchain
+
+    User/App->>MSP: Requests the file
+    MSP->>User/App: Returns the bytes plus<br>bucket root Merkle proof
+    User/App->>Blockchain: Checks proof against<br>file's on-chain commitment
+    Blockchain->>User/App: Confirms proof check passed,<br>returned data matches original upload
+```
+
+## Verify Storage Over Time
+
+Scheduled health checks, in the form of proof challenges, ensure BSPs retain your backup copies over time as follows:
```

Comment:
When does this process happen within the greater flow? Like is it periodically as defined by the user or is it baked into the process and done at an interval set by the network?

Reply:
As I understand it, this happens on a regular interval defined by the protocol.

Reply:
~~Resolving this so I can merge the PR~~ haha merged and unresolved for visibility